draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -230,8 +230,9 @@ happen through Unicast DNS as noted in {{RFC6762}}, Section 3.
 An ICE agent that supports mDNS candidates MUST support the situation where the
 hostname resolution results in more than one IP address. In this case, the ICE
 agent MUST take exactly one of the resolved IP addresses and ignore the others.
-The ICE agent SHOULD use the first IPv6 address resolved, if one exists, or
-the first IPv4 address, if not.
+The ICE agent MUST do this using an algorithm that consistently selects the same
+IP address, independent of the order of the IP addresses in the resolution
```

Comment:
TBH, I'm not sure how I would implement this. What did you have in mind here?

Reply:
The algorithm could be along the lines of:

- prefer IP addresses that share a subnet with a host candidate (most likely to succeed) 
   + for arbitration use fd00:: before 192.162, before 172.16, before 10.x (as per next rule)
   + prefer IPv6 over IPv4 (provided there is a host candidate with the same address family)
- prefer IP addresses with the longest-prefix-match with a host candidate
- after this maybe just use memcmp between destination addresses to order them

The key point is that for the same resolution result we should pick the same IP address, because otherwise you get inconsistent and confusing out comes (sometimes it works and sometimes it doesn't).

Reply:
If what we are after is repeatability, this seems fine. Maybe we should state the intent more clearly.
An implementation might for instance pick randomly the first time and keep in memory for future resolutions.
If what we are after is improved selection, we might want to be more specific about the rules.

Reply:
Agree we should just focus on repeatability, and avoid getting into prioritization (given that this should never happen).

Reply:
Changed to 'SHOULD ingnore'